.github/workflows/build-and-test.yml

```diff
@@ -21,9 +21,11 @@ jobs:
       SNYK_EXCLUSIONS: ${{ steps.snyk-exclusions.outputs.SNYK_EXCLUSIONS }}
 
     steps:
-      # Using Github checkout action to retrieve the codebase.
+      # Using Github checkout action to retrieve the pull request codebase.
```

Comment:
Isn't this workflow run in forked pull requests and regular branches?

Reply:
Yes, the workflow does run on all pull requests, but a few lines down, on line 28, we are specifically checking out the code tied to the pull request.  By changing the workflow trigger to `pull_request_target` it "runs in the context of the base of the pull request, rather than in the context of the merge commit, as the `pull_request` event does." ([events-that-trigger-workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target))

By checking out the PR code, it allows the code to be built and run unit tests and Snyk test on the updated code.  So it doesn't result in any difference for regular branches, but it allows the code from the forked repo pr to run and get tested.

Reply:
By adding the pr pull request specific stuff all the time, doesn't that mean that this will now *only* run for pull requests and not regular branches anymore?

Reply:
I don't think so, but I didn't test that out.  I'll give it a try and update once I find out.

Reply:
Updates have been made to get the workflow running for both forked and branch PRs.